clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -138,6 +142,23 @@
   }
 
   /**
+   * An alternative to construct a RemoteBlockInstream, bypassing all the checks.
+   * The returned RemoteBlockInStream can be used to call {#link #readRemoteByteBuffer}.
+   * TODO: remove this constructor, and modify all places using this constructor.
+   *
+   * @param file, any, could be null
+   * @param readType, any type
+   * @param blockIndex, any index
+   * @param ufsConf, any ufs configuration
+   * @param tachyonConf, any
+   * @param addFlag, add another field so that this constructor differentiates
+   */
+  public RemoteBlockInStream(TachyonFile file, ReadType readType, int blockIndex, Object ufsConf,
```

Comment:
Did a bit research, don't think it's possible..An alternative to this is to implement another static method which does the same thing, except the static one doesn't use the non-static members. Then we have to implement one in both `RemoteBlockInStream` and `RemoteBlockReader`, which is uglier.